examples/xgboost-quickstart/client.py

```diff
@@ -0,0 +1,162 @@
+import warnings
+from typing import Union
+from logging import INFO
+from datasets import Dataset, DatasetDict
+import xgboost as xgb
+
+import flwr as fl
+from flwr_datasets import FederatedDataset
+from flwr.common.logger import log
+from flwr.common import (
+    Code,
+    EvaluateIns,
+    EvaluateRes,
+    FitIns,
+    FitRes,
+    GetParametersIns,
+    GetParametersRes,
+    Parameters,
+    Status,
+)
+from flwr_datasets.partitioner import IidPartitioner
+
+
+warnings.filterwarnings("ignore", category=UserWarning)
+
+
+# Define data partitioning related functions
+def train_test_split(partition: Dataset, test_fraction: float, seed: int):
+    """Split the data into train and validation set given split rate."""
+    train_test = partition.train_test_split(test_size=test_fraction, seed=seed)
+    partition_train = train_test["train"]
+    partition_test = train_test["test"]
+
+    num_train = len(partition_train)
+    num_test = len(partition_test)
+
+    return partition_train, partition_test, num_train, num_test
+
+
+def transform_dataset_to_dmatrix(data: Union[Dataset, DatasetDict]) -> xgb.core.DMatrix:
+    """Transform dataset to DMatrix format for xgboost."""
+    x = data["inputs"]
+    y = data["label"]
+    new_data = xgb.DMatrix(x, label=y)
+    return new_data
+
+
+# Load (HIGGS) dataset and conduct partitioning
+partitioner = IidPartitioner(num_partitions=10)
+fds = FederatedDataset(dataset="jxie/higgs", partitioners={"train": partitioner})
+
+# Let's use the first partition as an example
+partition = fds.load_partition(idx=0, split="train")
```

Comment:
Should we add the possibility to load a specific partition by providing `--node-id 0` or `--node-id 1`? And then use `IidPartitioner(num_partitions=2)` above?

Reply:
ok, i'll pull back the arguments parser only for `--node-id`